adapters/repos/db/vector/flat/index.go

```diff
@@ -0,0 +1,543 @@
+}
+
+func (index *flat) UpdateUserConfig(updated schema.VectorIndexConfig, callback func()) error {
+	parsed, ok := updated.(flatent.UserConfig)
+	if !ok {
+		callback()
+		return errors.Errorf("config is not UserConfig, but %T", updated)
+	}
+
+	// Store automatically as a lock here would be very expensive, this value is
+	// read on every single user-facing search, which can be highly concurrent
+	atomic.StoreInt64(&index.ef, int64(parsed.EF))
+
+	callback()
+	return nil
+}
+
+func (index *flat) Drop(ctx context.Context) error {
+	// nothing to do here
+	// Shard::drop will take care of handling store's buckets
+	return nil
+}
+
+func (index *flat) Flush() error {
+	// nothing to do here
+	// Shard will take care of handling store's buckets
+	return nil
+}
+
+func (index *flat) Shutdown(ctx context.Context) error {
+	// nothing to do here
+	// Shard::shutdown will take care of handling store's buckets
+	return nil
+}
+
+func (index *flat) SwitchCommitLogs(context.Context) error {
+	return nil
+}
+
+func (index *flat) ListFiles(ctx context.Context, basePath string) ([]string, error) {
+	// nothing to do here
+	// Shard::listBackupFiles will take care of handling store's buckets
+	return []string{}, nil
+}
+
+func (i *flat) ValidateBeforeInsert(vector []float32) error {
+	return nil
+}
+
+func (index *flat) PostStartup() {
+}
+
+func (index *flat) Dump(labels ...string) {
+	if len(labels) > 0 {
+		fmt.Printf("--------------------------------------------------\n")
+		fmt.Printf("--  %s\n", strings.Join(labels, ", "))
+	}
+	fmt.Printf("--------------------------------------------------\n")
+	fmt.Printf("ID: %s\n", index.id)
+	fmt.Printf("--------------------------------------------------\n")
+}
+
+func (index *flat) DistanceBetweenVectors(x, y []float32) (float32, bool, error) {
+	return index.distancerProvider.SingleDist(x, y)
+}
+
+func (index *flat) ContainsNode(id uint64) bool {
+	return true
+}
+
+func (index *flat) DistancerProvider() distancer.Provider {
+	return index.distancerProvider
+}
+
+func newSearchByDistParams(maxLimit int64) *common.SearchByDistParams {
+	initialOffset := 0
+	initialLimit := common.DefaultSearchByDistInitialLimit
+
+	return common.NewSearchByDistParams(initialOffset, initialLimit, initialOffset+initialLimit, maxLimit)
+}
+
+func ValidateUserConfigUpdate(initial, updated schema.VectorIndexConfig) error {
+	/*
+		initialParsed, ok := initial.(flatent.UserConfig)
+		if !ok {
+			return errors.Errorf("initial is not UserConfig, but %T", initial)
+		}
+
+		updatedParsed, ok := updated.(flatent.UserConfig)
+		if !ok {
+			return errors.Errorf("updated is not UserConfig, but %T", updated)
+		}
+
+		immutableFields := []immutableParameter{
+			{
+				name:     "distance",
+				accessor: func(c flatent.UserConfig) interface{} { return c.Distance },
+			},
+		}
+
+		for _, u := range immutableFields {
+			if err := validateImmutableField(u, initialParsed, updatedParsed); err != nil {
+				return err
+			}
+		}
+	*/
```

Comment:
Just to be sure, this commented part will implemented in later PR's right?

Reply:
@abdelr? should we remove that?

Reply:
Already removed